### PR TITLE
First draft of rfc9

### PR DESCRIPTION
**Pull request by jandrieu**

### CODE REVIEW

**Comment by jandrieu** [comment]
@ig-shaun Please review and merge.

File: nft_rfc_009.md [hunk 1]
```diff
@@ -0,0 +1,965 @@
...
+author: Joe Andrieu @jandrieu
+created: 2021-03-18
+modified: 
+---
+
+Abstract
+========
+
+Interchain Identifiers (IIDs) are a family of DID methods for
+blockchain-based assets such as tokens, smart contracts, and addresses.
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
specifically designed to refer to on-chain assets, such as NFTs, fungible tokens, namespace records and wallets.
```

File: nft_rfc_009.md [hunk 2]
```diff
@@ -0,0 +1,965 @@
...
+Built on Decentralized Identifiers from the World Wide Web Consortium
+[[1]](#ref1), IIDs are 100% conformant DIDs; IIDs ***are*** DIDs and IID
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
[[1]](#ref1), IIDs are fully conformant DIDs and therefore ***are*** DIDs. IID
```

File: nft_rfc_009.md [hunk 3]
```diff
@@ -0,0 +1,965 @@
...
+documents are DID documents. The IID specification also provides
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
documents are DID documents. The IID specification provides
```

File: nft_rfc_009.md [hunk 4]
```diff
@@ -0,0 +1,965 @@
...
+additional features designed for on-chain assets, such as the ability to
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
additional features which are purpose-designed for describing and interacting with on-chain assets. Such as Linked Resources, which enable
```

File: nft_rfc_009.md [hunk 5]
```diff
@@ -0,0 +1,965 @@
...
+verifiably yet privately associate arbitrary digital or real-world
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
arbitrary digital or real-world resources to be verifiably, but privately, associated with
```

File: nft_rfc_009.md [hunk 6]
```diff
@@ -0,0 +1,965 @@
...
+assets with the underlying IID Subject. IIDs also constrain compatible
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
the digital asset, which is the IID Subject.
```

File: nft_rfc_009.md [hunk 7]
```diff
@@ -0,0 +1,965 @@
...
+DID methods in particular ways, including a restriction on the DID
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
When referring to an IID, it is implicit that this identifies an on-chain asset. This constraint has the advantage that IID's are self-referential. Which overcomes the HTTP Range14 problem of DIDs which refer to resources that cannot be located at an end-point, such as a physical object. DID Methods which conform to this IID specification MUST resolve to a uniquely identified digital asset, within a unique blockchain namespace.
```

File: nft_rfc_009.md [hunk 8]
```diff
@@ -0,0 +1,965 @@
...
+Subject---IID Subjects MUST be on-chain resources. Any DID method that
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion

```

File: nft_rfc_009.md [hunk 9]
```diff
@@ -0,0 +1,965 @@
...
+meets the requirements in this specification ***is*** an IID method, and
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion

```

File: nft_rfc_009.md [hunk 10]
```diff
@@ -0,0 +1,965 @@
...
+its IIDs ***are*** DIDs. Any DID software that is conformant to the W3C
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
Any software application using DIDs which conform to the W3C DID-Core
```

File: nft_rfc_009.md [hunk 11]
```diff
@@ -0,0 +1,965 @@
...
+specification SHOULD be able to work with IIDs and IID documents,
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
specification SHOULD be able to operate with IIDs and IID documents.
```

File: nft_rfc_009.md [hunk 12]
```diff
@@ -0,0 +1,965 @@
...
+although some IID-specific features may require additional support.
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
Although some IID-specific features may require additional support. For instance, to dereference Linked Resources.
```

File: nft_rfc_009.md [hunk 13]
```diff
@@ -0,0 +1,965 @@
...
+
+Introduction
+============
+
+Identifiers are needed to refer to both on-chain and off-chain
+resources. Non-fungible tokens (NFTs) require a cryptographically secure
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
resources. On-chain digital assets require a cryptographically secure
```

File: nft_rfc_009.md [hunk 14]
```diff
@@ -0,0 +1,965 @@
...
+means to both refer to specific tokens and prove control or ownership
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
mechanisms for referring to the asset and to prove control or ownership of the asset,
```

File: nft_rfc_009.md [hunk 15]
```diff
@@ -0,0 +1,965 @@
...
+without reliance on a trusted third party.
+
```
Comment (ig-shaun) [inline nft_rfc_009.md:39]: All on-chain digital assets MUST have a unique identifier which can be dereferenced to locate the asset as a resource located in a specific chain namespace.

File: nft_rfc_009.md [hunk 16]
```diff
@@ -0,0 +1,965 @@
...
+We refer to this new class of identifiers as Interchain Identifiers, or
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
We refer to this new class of identifiers as Interchain Identifiers (IIDs).
```

File: nft_rfc_009.md [hunk 17]
```diff
@@ -0,0 +1,965 @@
...
+IIDs for short. We propose this new identifier would be appropriate for
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
Interchain Identifiers SHOULD be appropriate for
```

File: nft_rfc_009.md [hunk 18]
```diff
@@ -0,0 +1,965 @@
...
+any on-chain asset, including smart contracts, fungible tokens, and
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
identifying and referring to any on-chain asset. Including smart contracts, fungible tokens,
```

File: nft_rfc_009.md [hunk 19]
```diff
@@ -0,0 +1,965 @@
...
+non-fungible tokens.
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
non-fungible tokens, namespace records, and wallets.
```

File: nft_rfc_009.md [hunk 20]
```diff
@@ -0,0 +1,965 @@
...
+
+IIDs build on a foundation of global identifiers dating back to Uniform
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
IIDs build on a foundation of Internet identifiers, dating back to Uniform
```

File: nft_rfc_009.md [hunk 21]
```diff
@@ -0,0 +1,965 @@
...
+Resource Locators and Uniform Resource Identifiers [[2]](#ref2), which are the
+foundation of the World Wide Web. URLs and URIs allow for different
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
foundation of the World Wide Web. URIs allow for different
```

File: nft_rfc_009.md [hunk 22]
```diff
@@ -0,0 +1,965 @@
...
+schemes that specify particular mechanisms for interpreting and applying
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
to specify the mechanisms for interpreting and applying
```

File: nft_rfc_009.md [hunk 23]
```diff
@@ -0,0 +1,965 @@
...
+the rest of the identifier. For example,
+```http://example.com``` specifies a web-based resource that can be 
+retrieved using the hypertext transfer protocol (http) and ```
+mailto:joe@example.com``` specifies a resource that can receive email
+messages using the SMTP protocol.
+
+IIDs are Decentralized Identifiers (DIDs), which build on URIs and URLs 
+to provide a class of identifiers one can verify without reliance on a
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
to provide a class of identifiers which can verified without reliance on a
```

File: nft_rfc_009.md [hunk 24]
```diff
@@ -0,0 +1,965 @@
...
+trusted third party. DIDs specifically support cryptographic
+verification methods like public/private key cryptography on any number
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
verification methods, such as public-private cryptographic keys derived from any number
```

File: nft_rfc_009.md [hunk 25]
```diff
@@ -0,0 +1,965 @@
...
+of curves, which in turn are used for verification relationships like
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
of curves. These keys are used to establish verification relationships for the purpose of
```

File: nft_rfc_009.md [hunk 26]
```diff
@@ -0,0 +1,965 @@
...
+authentication, assertion, capability delegation, and capability
+invocation. When one uses a DID for those relationships, a verifying
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
invocation. A DID Method establishes these verification relationships in a way that each cryptographic method is associated with its specific purpose. 
```

File: nft_rfc_009.md [hunk 27]
```diff
@@ -0,0 +1,965 @@
...
+retrieved using the hypertext transfer protocol (http) and ```
+mailto:joe@example.com``` specifies a resource that can receive email
+messages using the SMTP protocol.
+
+IIDs are Decentralized Identifiers (DIDs), which build on URIs and URLs 
+to provide a class of identifiers one can verify without reliance on a
+trusted third party. DIDs specifically support cryptographic
+verification methods like public/private key cryptography on any number
+of curves, which in turn are used for verification relationships like
+authentication, assertion, capability delegation, and capability
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
authentication, assertion, capability delegation, or capability
```

File: nft_rfc_009.md [hunk 28]
```diff
@@ -0,0 +1,965 @@
...
+invocation. When one uses a DID for those relationships, a verifying
+party can use cryptographic material associated with the DID to verify
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion

```

File: nft_rfc_009.md [hunk 29]
```diff
@@ -0,0 +1,965 @@
...
+the use is appropriate. DID-URLs extend the base DID syntax to support
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
DID-URLs extend the base DID syntax to support 
```

File: nft_rfc_009.md [hunk 30]
```diff
@@ -0,0 +1,965 @@
...
+to provide a class of identifiers one can verify without reliance on a
+trusted third party. DIDs specifically support cryptographic
+verification methods like public/private key cryptography on any number
+of curves, which in turn are used for verification relationships like
+authentication, assertion, capability delegation, and capability
+invocation. When one uses a DID for those relationships, a verifying
+party can use cryptographic material associated with the DID to verify
+the use is appropriate. DID-URLs extend the base DID syntax to support 
+```/path```, ```?query```, and ```#fragment``` parts. IIDs use the 
+```/path``` and ```#fragment``` parts of URLs to refer to IID References
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
```/path``` and ```#fragment``` parts of URLs for IID References 
```

File: nft_rfc_009.md [hunk 31]
```diff
@@ -0,0 +1,965 @@
...
+of curves, which in turn are used for verification relationships like
+authentication, assertion, capability delegation, and capability
+invocation. When one uses a DID for those relationships, a verifying
+party can use cryptographic material associated with the DID to verify
+the use is appropriate. DID-URLs extend the base DID syntax to support 
+```/path```, ```?query```, and ```#fragment``` parts. IIDs use the 
+```/path``` and ```#fragment``` parts of URLs to refer to IID References 
+and IID Resources.
+
+We describe how IIDs and DIDs relate in section [IIDs as DIDs](#IIDs-as-DIDs).
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
The relationship between IIDs and DIDs is further described in the [IIDs as DIDs](#IIDs-as-DIDs) section.
```

File: nft_rfc_009.md [hunk 32]
```diff
@@ -0,0 +1,965 @@
...
+
+IIDs also provide a consistent, verifiable approach for referring to
+off-chain resources, whether digital, real-world, or conceptual. In this
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
off-chain resources. Whether these resources are digital, physical, or conceptual. To achieve this
```

File: nft_rfc_009.md [hunk 33]
```diff
@@ -0,0 +1,965 @@
...
+role, IIDs rely on content-based identifiers and content-based
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
IID methods SHOULD use content-based identifiers and content-based
```

File: nft_rfc_009.md [hunk 34]
```diff
@@ -0,0 +1,965 @@
...
+addressing systems, where a hash of the digital asset is used to
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
addressing systems, in which a hash of linked resource is used to
```

File: nft_rfc_009.md [hunk 35]
```diff
@@ -0,0 +1,965 @@
...
+unambiguously verify the asset is the intended resource. Standards like
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
unambiguously verify the resource. Standards such as
```

File: nft_rfc_009.md [hunk 36]
```diff
@@ -0,0 +1,965 @@
...
+multihash [\[3\]](#ref3) provide for a wide range of different hash algorithms
+and representations. Approaches using merkle trees can turn hashes into
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
and representations.
```

File: nft_rfc_009.md [hunk 37]
```diff
@@ -0,0 +1,965 @@
...
+a hashgraphs, allowing an arbitrary number of resources to be
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
Using mechanisms such as Hashgraphs enables an arbitrary number of verifiable resources to be linked to a digital asset, without even revealing the number of resources.
```

File: nft_rfc_009.md [hunk 38]
```diff
@@ -0,0 +1,965 @@
...
+independently verified without revealing even the number of those assets
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
```

File: nft_rfc_009.md [hunk 39]
```diff
@@ -0,0 +1,965 @@
...
+publicly. These content identifiers can provide robust verifiability no
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
Content Identifiers (CIDs) provide robust verifiability and enable resources to be addressed without path-dependency. These mechanisms enable rich data to be associated with digital resources, without compromising privacy.
```

File: nft_rfc_009.md [hunk 40]
```diff
@@ -0,0 +1,965 @@
...
+matter where such resources might be located or retrieved.
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
```

File: nft_rfc_009.md [hunk 41]
```diff
@@ -0,0 +1,965 @@
...
+
+Specifically for NFTs, IIDs allow you to
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
Implementing IIDs allows tokenized digital assets to be:
```

File: nft_rfc_009.md [hunk 42]
```diff
@@ -0,0 +1,965 @@
...
+
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
```

File: nft_rfc_009.md [hunk 43]
```diff
@@ -0,0 +1,965 @@
...
+1. Identify individual NFTs as well as their token class
+
```
Comment (ig-shaun) [inline nft_rfc_009.md:73]:
```suggestion
```

File: nft_rfc_009.md [hunk 44]
```diff
@@ -0,0 +1,965 @@
...
+multihash [\[3\]](#ref3) provide for a wide range of different hash algorithms
+and representations. Approaches using merkle trees can turn hashes into
+a hashgraphs, allowing an arbitrary number of resources to be
+independently verified without revealing even the number of those assets
+publicly. These content identifiers can provide robust verifiability no
+matter where such resources might be located or retrieved.
+
+Specifically for NFTs, IIDs allow you to
+
+1. Identify individual NFTs as well as their token class
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
1. Identified as unique digital assets that belong to a specific class of tokens. 
```

File: nft_rfc_009.md [hunk 45]
```diff
@@ -0,0 +1,965 @@
...
+
+2. Identify resources associated with that NFT
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
2. Linked to verifiable on-chain and off-chain resources.
```

File: nft_rfc_009.md [hunk 46]
```diff
@@ -0,0 +1,965 @@
...
+
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
```

File: nft_rfc_009.md [hunk 47]
```diff
@@ -0,0 +1,965 @@
...
+3. Exercise rights related to those resources using cryptography anchored to the NFT
+
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
```

File: nft_rfc_009.md [hunk 48]
```diff
@@ -0,0 +1,965 @@
...
+publicly. These content identifiers can provide robust verifiability no
+matter where such resources might be located or retrieved.
+
+Specifically for NFTs, IIDs allow you to
+
+1. Identify individual NFTs as well as their token class
+
+2. Identify resources associated with that NFT
+
+3. Exercise rights related to those resources using cryptography anchored to the NFT
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
3. Accorded rights in relation to both the token and its linked resources. Including executable rights, which may be issued and delegated off-chain using a mechanism such as Authorization Capabilities (zCaps), to access, invoke and transact with specific on-chain or off-chain services.
```

File: nft_rfc_009.md [hunk 49]
```diff
@@ -0,0 +1,965 @@
...
+
+IIDs allow these functions a manner that addresses the privacy
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
4. Afforded privacy in relation to the token's metadata. Including what the token represents, the associated services which can be invoked, and the resources which are linked to the token.
```

File: nft_rfc_009.md [hunk 50]
```diff
@@ -0,0 +1,965 @@
...
+requirements for such references, with flexible support for use cases
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
```

File: nft_rfc_009.md [hunk 51]
```diff
@@ -0,0 +1,965 @@
...
+that need to make different choices. Our approach offers a robust,
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
```

File: nft_rfc_009.md [hunk 52]
```diff
@@ -0,0 +1,965 @@
...
+flexible, and capable identifier mechanism for NFTs, and on-chain
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
```

File: nft_rfc_009.md [hunk 53]
```diff
@@ -0,0 +1,965 @@
...
+assets, of all types.
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
```

File: nft_rfc_009.md [hunk 54]
```diff
@@ -0,0 +1,965 @@
...
+
+Use Cases
+=========
+
+This specification was designed for Non-Fungible Tokens. NFT-RFC-002
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
The IID specification was designed to address the requirements for Non-Fungible Token (NFT) use-cases, which are described in NFT-RFC-002.
```

File: nft_rfc_009.md [hunk 55]
```diff
@@ -0,0 +1,965 @@
...
+[[4]](#ref4) describes the range of types of tokens as well as several detailed
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
[[4]](#ref4).
```

File: nft_rfc_009.md [hunk 56]
```diff
@@ -0,0 +1,965 @@
...
+user stories covered by this work. NFT-RFC-008 [[5]](#ref5) describes how one
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
The specific use-case of tokenizing Renewable Energy Certificates (RECs), described in NFT-RFC-008 [[5]](#ref5) demonstrates how IIDs can be used together with DID-related W3C specifications for
```

File: nft_rfc_009.md [hunk 57]
```diff
@@ -0,0 +1,965 @@
...
+might apply IIDs, Verifiable Credentials, and Authorization Capabilities to define,
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
Verifiable Credentials (VCs) and Authorization Capabilities (zCaps), to define a class of tokens (REC-2021),
```

File: nft_rfc_009.md [hunk 58]
```diff
@@ -0,0 +1,965 @@
...
+authorize, issue, use, and sell NFT-based Renewable Energy Certificates
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
authorize the issuance of verifiable credentials, verify claims of renewable energy production, and then mint and trade Renewable Energy Certificates as NFTs. With accorded rights and linked resources.
```

File: nft_rfc_009.md [hunk 59]
```diff
@@ -0,0 +1,965 @@
...
+(RECs) according to an REC 2021 standard.
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
```

File: nft_rfc_009.md [hunk 60]
```diff
@@ -0,0 +1,965 @@
...
+
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
```

File: nft_rfc_009.md [hunk 61]
```diff
@@ -0,0 +1,965 @@
...
+Throughout this document we use the REC2021 use case for examples. In summary:
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
Throughout this document we use the REC2021 use case for examples. 
In summary:
```

File: nft_rfc_009.md [hunk 62]
```diff
@@ -0,0 +1,965 @@
...
+2. The UN publishes that token template, resuling in an IID for the new class.
+2. The UN uses a WebKMS system so that it can delegate the ability to sign Verifiable
+Credentials.
+3. The UN defines a vocabulary for Certification Reports and Verification Reports as
+Verifiable Credentials
+4. The UN delegates the ability to create Verification Reports (with certain constraints)
+to known, qualified verifiers.
+5. The UN delegates
+
+   a) the ability to create Certification Reports (with certain
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
   a) the capability to create Certification Reports (with certain
```

File: nft_rfc_009.md [hunk 63]
```diff
@@ -0,0 +1,965 @@
...
+constraints) and
+
+   b) the ability to mint REC2021 tokens (with certain constraints)
```
Comment (ig-shaun) [inline nft_rfc_009.md]:
```suggestion
   b) the capability to mint REC2021 tokens (with certain constraints)
```

**Review by ig-shaun — CHANGES_REQUESTED** [review]

File: nft_rfc_009.md [hunk 64]
```diff
@@ -0,0 +1,951 @@
...
+Although some IID-specific features may require additional support. For instance, to dereference Linked Resources.
+
+Introduction
+============
+
+Identifiers are needed to refer to both on-chain and off-chain
+resources. On-chain digital assets require a cryptographically secure
+mechanisms for referring to the asset and to prove control or ownership of the asset,
+without reliance on a trusted third party.
+
```
Comment (jandrieu) [inline nft_rfc_009.md]:
```suggestion
All on-chain digital assets MUST have a unique identifier which can be dereferenced to locate the asset as a resource located in a specific chain namespace.
```

**Pull request merged**